packages/core/content-manager/server/controllers/collection-types.js

```diff
@@ -181,6 +181,64 @@ module.exports = {
     ctx.body = await permissionChecker.sanitizeOutput(result);
   },
 
+  async bulkPublish(ctx) {
+    const { userAbility } = ctx.state;
+    const { model } = ctx.params;
+    const { query, body } = ctx.request;
+    const { ids } = body;
+
+    await validateBulkActionInput(body);
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.publish()) {
+      return ctx.forbidden();
+    }
+
+    const permissionQuery = await permissionChecker.sanitizedQuery.publish(query);
+
+    const idsWhereClause = { id: { $in: ids } };
+    const params = {
+      ...permissionQuery,
+      filters: {
+        $and: [idsWhereClause].concat(permissionQuery.filters || []),
+      },
+    };
```

Comment:
I am not quite sure this is enough for permission checking, I think we should also use:
```js
    if (permissionChecker.cannot.publish(entity)) {
      return ctx.forbidden();
    }
```

I am checking now, but with the current implementation I am not sure [custom conditions ](https://docs.strapi.io/dev-docs/configurations/rbac#declaring-new-conditions) would work: